keps/sig-auth/3130-kms-observability/README.md

```diff
@@ -0,0 +1,224 @@
+
+The UID generated in the kube-apiserver will be used:
+
+1. For logging in the kube-apiserver. All envelope operations to the kms-plugin will be logged with the corresponding UID.
+2. Sent to the kms-plugin as part of the `EncryptRequest` and `DecryptRequest` structs.
+
+### Test Plan
+
+Unit tests covering:
+
+1. Generation of UID for each envelope operation
+
+Integration test covering:
+
+1. Logging of UID in kube-apiserver
+2. UID in the `EncryptRequest` and `DecryptRequest`
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+-->
+
+- [x] Other
+  - Describe the mechanism: The UID field will be added by default to the `EncryptRequest` and `DecryptRequest` structs in the kms-plugin interface. For older Kubernetes versions where UID isn't generated, the UID field will be empty.
+
+###### Does enabling the feature change any default behavior?
+
+No change in the default behavior.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+The enhancement is centered around better logging for observability and ability to correlate the sequence of calls that are involved in the enveloping operation. It is not possible to roll back the UID generation.
+
+### Monitoring Requirements
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other (treat as last resort)
+  - Details: Logs in kube-apiserver, kms-plugin and KMS will be logged with the corresponding UID.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+There should be no impact on the SLO with this change.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Other (treat as last resort)
+  - Details: Logs in kube-apiserver, kms-plugin and KMS will be logged with the corresponding UID.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+- Secret data encryption with external kms-plugin is unavailable
```

Comment:
To me this seems more appropriate to the KMS feature as a whole but if we talking about KMS observability only, I feel like the following questions would be more appropriate to elaborate on:

- How does this feature react if the KMS provider / KMS is unavailable?
- How does this feature react if encrypted resources are unavailable?

To answer that question we could elaborate on how to use the various information introduced by this KEP to troubleshoot these problems.